File: fix_encoding.js
const fs = require("fs");
let content = fs.readFileSync("crates/codex-plus-core/src/app_paths.rs","utf8");

// Scan for any non-printable/non-ASCII bytes that aren't valid UTF-8
let lines = content.split("\n");

// Rebuild the macos_app_candidates function cleanly
for(let i=0;i<lines.length;i++){
    if(lines[i].includes("fn macos_app_candidates")){
        // Collect the function
        let fnLines = [];
        let j = i;
        while(j < lines.length && lines[j].trim() !== "}"){
            fnLines.push(lines[j]);
            j++;
        }
        fnLines.push(lines[j]); // closing }
        console.log("Found function at L"+(i+1)+" to L"+(j+1));
        console.log(fnLines.join("\n"));
        
        // Replace the function body with clean code
        let cleanFn = [
            "#[cfg(target_os = \"macos\")]",
            "fn macos_app_candidates(root: &Path) -> Vec<PathBuf> {",
            "    if root.extension() == Some(OsStr::new(\"app\")) {",
            "        return vec![root.to_path_buf()];",
            "    }",
            "    vec![\"Codex.app\", \"OpenAI Codex.app\", \"OpenAI.Codex.app\"]",
            "        .into_iter()",
            "        .map(|name| root.join(name))",
            "        .collect()",
            "}"
        ];
        
        // Replace lines
        lines.splice(i, j-i+1, ...cleanFn);
        console.log("Replaced with clean function");
        break;
    }
}

fs.writeFileSync("crates/codex-plus-core/src/app_paths.rs", lines.join("\n"), "utf8");
console.log("Done");

File: scripts/fix_encoding.py
import sys
d = open("J:/codex-work/LDCodex/crates/codex-plus-core/src/app_paths.rs","rb").read()

# Remove all non-ASCII bytes
cleaned = bytearray()
for b in d:
    if b < 128:
        cleaned.append(b)
d = bytes(cleaned)

# Convert CRLF to LF
d = d.replace(b"\r\n", b"\n")

# Fix the }#[ one-liner
lines = d.split(b"\n")
fixed = []
for i, line in enumerate(lines):
    if b"}#[cfg(target_os" in line:
        fixed.append(b"}")
        fixed.append(line[line.index(b"#[cfg"):])
    else:
        fixed.append(line)

d = b"\n".join(fixed)
d = d.rstrip(b"\n") + b"\n"

# Remove const CODEX_PREFIX line
lines = d.split(b"\n")
new_lines = [line for line in lines if not line.strip().startswith(b"const CODEX_PREFIX")]
d = b"\n".join(new_lines)
d = d.rstrip(b"\n") + b"\n"

open("J:/codex-work/LDCodex/crates/codex-plus-core/src/app_paths.rs","wb").write(d)
print("Done. Size:", len(d))
